roles/kubernetes-apps/network_plugin/contiv/tasks/main.yml

```diff
@@ -0,0 +1,118 @@
+- name: Create contiv-etcd-proxy DaemonSet
+  run_once: true
+  kube:
+    name: "contiv-etcd-proxy"
+    kubectl: "{{bin_dir}}/kubectl"
+    filename: "{{kube_config_dir}}/contiv-etcd-proxy.yaml"
+    resource: "daemonset"
+    namespace: "{{system_namespace}}"
+
+- name: Create contiv-netplugin DaemonSet
+  run_once: true
+  kube:
+    name: "contiv-netplugin"
+    kubectl: "{{bin_dir}}/kubectl"
+    filename: "{{kube_config_dir}}/contiv-netplugin.yaml"
+    resource: "daemonset"
+    namespace: "{{system_namespace}}"
+
+- name: Create contiv-netmaster DaemonSet
+  run_once: true
+  kube:
+    name: "contiv-netmaster"
+    kubectl: "{{bin_dir}}/kubectl"
+    filename: "{{kube_config_dir}}/contiv-netmaster.yaml"
+    resource: "daemonset"
+    namespace: "{{system_namespace}}"
+
+- name: Create contiv-api-proxy DaemonSet
+  run_once: true
+  kube:
+    name: "contiv-api-proxy"
+    kubectl: "{{bin_dir}}/kubectl"
+    filename: "{{kube_config_dir}}/contiv-api-proxy.yaml"
+    resource: "daemonset"
+    namespace: "{{system_namespace}}"
+  when: "{{ contiv_enable_api_proxy }}"
+
+- name: Create contiv-config ConfigMap
+  run_once: true
+  kube:
+    name: "contiv-config"
+    kubectl: "{{bin_dir}}/kubectl"
+    filename: "{{kube_config_dir}}/contiv-config.yaml"
+    resource: "configmap"
+    namespace: "{{system_namespace}}"
+
+- name: Create contiv ClusterRoleBinding
+  run_once: true
+  kube:
+    name: "contiv-{{item}}"
+    kubectl: "{{bin_dir}}/kubectl"
+    filename: "{{kube_config_dir}}/contiv-{{item}}-clusterrolebinding.yaml"
+    resource: "clusterrolebinding"
+    namespace: "{{system_namespace}}"
+  with_items:
+    - netmaster
+    - netplugin
+
+- name: Create contiv ClusterRole
+  run_once: true
+  kube:
+    name: "contiv-{{item}}"
+    kubectl: "{{bin_dir}}/kubectl"
+    filename: "{{kube_config_dir}}/contiv-{{item}}-clusterrole.yaml"
+    resource: "clusterrole"
+    namespace: "{{system_namespace}}"
+  with_items:
+    - netmaster
+    - netplugin
+
+- name: Create contiv ServiceAccount
+  run_once: true
+  kube:
+    name: "contiv-{{item}}"
+    kubectl: "{{bin_dir}}/kubectl"
+    filename: "{{kube_config_dir}}/contiv-{{item}}-serviceaccount.yaml"
+    resource: "serviceaccount"
+    namespace: "{{system_namespace}}"
+  with_items:
+    - netmaster
+    - netplugin
+
+
+- name: Contiv | Wait for netmaster
+  command: |
+    {{ bin_dir }}/netctl --netmaster "http://127.0.0.1:{{ contiv_netmaster_port }}" tenant ls
+  register: tenant_result
+  until: tenant_result.stdout.find("default") != -1
+  retries: 9
+  delay: 10
+  run_once: true
+
+- name: Contiv | Check if default-net exists
+  command: |
+    {{ bin_dir }}/netctl --netmaster "http://127.0.0.1:{{ contiv_netmaster_port }}" net ls
+  register: net_result
+  run_once: true
+
+- name: Contiv | Set global configuration
```

Comment:
Does this ultimately write to etcd? It might be easier for upgrades handling if you wrote this config into etcd directly before creating the kubernetes resources.